modules/contact/src/actions/ContactAction.C

```diff
@@ -236,62 +244,64 @@ ContactAction::act()
-            getParam<VariableName>("secondary_gap_offset")};
-      if (isParamValid("mapped_primary_gap_offset"))
-        params.set<std::vector<VariableName>>("mapped_primary_gap_offset") = {
-            getParam<VariableName>("mapped_primary_gap_offset")};
-
-      params.set<bool>("use_displaced_mesh") = true;
-      std::string name = _name + "_contact_" + Moose::stringify(contact_auxkernel_counter);
-
-      _problem->addAuxKernel("PenetrationAux", name, params);
-    }
-    // Add ContactPressureAuxAction
-    {
-      InputParameters params = _factory.getValidParams("ContactPressureAux");
-      params.applyParameters(parameters(), {"order"});
-
-      std::vector<VariableName> displacements =
-          getParam<std::vector<VariableName>>("displacements");
-      const auto order = _problem->systemBaseNonlinear()
-                             .system()
-                             .variable_type(displacements[0])
-                             .order.get_order();
-
-      params.set<MooseEnum>("order") = Utility::enum_to_string<Order>(OrderWrapper{order});
-      params.set<std::vector<BoundaryName>>("boundary") = {_secondary};
-      params.set<BoundaryName>("paired_boundary") = _primary;
-      params.set<AuxVariableName>("variable") = "contact_pressure";
-      params.addRequiredCoupledVar("nodal_area", "The nodal area");
-      params.set<std::vector<VariableName>>("nodal_area") = {"nodal_area_" + _name};
+      {
+        InputParameters params = _factory.getValidParams("PenetrationAux");
+        params.applyParameters(parameters(),
+                               {"secondary_gap_offset", "mapped_primary_gap_offset", "order"});
+
+        std::vector<VariableName> displacements =
+            getParam<std::vector<VariableName>>("displacements");
+        const auto order = _problem->systemBaseNonlinear()
+                               .system()
+                               .variable_type(displacements[0])
+                               .order.get_order();
+
+        params.set<MooseEnum>("order") = Utility::enum_to_string<Order>(OrderWrapper{order});
+        params.set<ExecFlagEnum>("execute_on") = {EXEC_INITIAL, EXEC_LINEAR};
+        params.set<std::vector<BoundaryName>>("boundary") = {_secondary[contact_pair]};
+        params.set<BoundaryName>("paired_boundary") = _primary[contact_pair];
+        params.set<AuxVariableName>("variable") = "penetration";
+        if (isParamValid("secondary_gap_offset"))
+          params.set<std::vector<VariableName>>("secondary_gap_offset") = {
+              getParam<VariableName>("secondary_gap_offset")};
+        if (isParamValid("mapped_primary_gap_offset"))
+          params.set<std::vector<VariableName>>("mapped_primary_gap_offset") = {
+              getParam<VariableName>("mapped_primary_gap_offset")};
+        params.set<bool>("use_displaced_mesh") = true;
+        std::string name = _name + "_contact_" + Moose::stringify(contact_auxkernel_counter);
```

Comment:
Would we have a same name for different boundary pairs? 

It might be good if we can set different names of those auxiliary objects for different pairs.

Reply:
Here, `contact_auxkernel_counter` is a static counter that gets incremented for each contact pair. The name should be changing for each contact pair.

Reply:
I see. I missed `++` operation :-)